### PR TITLE
feat: upgrade to sentry 2024.02 and bump python to 3.11

### PR DESCRIPTION
**Pull request by dlouzan**

### CODE REVIEW

**Comment by dlouzan** [comment]
@max-wittig @nejch Any idea how to solve this problem on the pipeline? Apparently the egg format for extras is not supported anymore, so it's not installing the whole development dependencies suite:

```
pip3 install -e git+https://github.com/getsentry/sentry.git@24.2.0#egg=sentry[dev]

DEPRECATION: git+https://github.com/getsentry/sentry.git@24.2.0#egg=sentry[dev] contains an egg fragment with a non-PEP 508 name pip 25.0 will enforce this behaviour change. A possible replacement is to use the req @ url syntax, and remove the egg fragment. Discussion can be found at https://github.com/pypa/pip/issues/11617

Obtaining sentry[dev] from git+https://github.com/getsentry/sentry.git@24.2.0#egg=sentry[dev]
  Cloning https://github.com/getsentry/sentry.git (to revision 24.2.0) to ./venv/src/sentry
  Running command git clone --filter=blob:none --quiet https://github.com/getsentry/sentry.git /home/runner/work/sentry-auth-oidc/sentry-auth-oidc/venv/src/sentry
  Running command git checkout -q cfcff1f4fda857dcddc9401efb689b6af92a3214
  Resolved https://github.com/getsentry/sentry.git to commit cfcff1f4fda857dcddc9401efb689b6af92a3214
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Checking if build backend supports build_editable: started
  Checking if build backend supports build_editable: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'

WARNING: sentry 24.2.0 does not provide the extra 'dev'
```

**Comment by max-wittig** [comment]
Seems like maybe we can just remove the extra part, like here: https://github.com/getsentry/sentry-plugins/blob/master/Makefile#L26

From the logs:
`WARNING: sentry 24.2.0 does not provide the extra 'dev'`

**Comment by dlouzan** [comment]
@nejch Yes, that'd be a very good idea, we need to adapt this fully as upstream completely killed any kind of support for server-side plugins.

**Comment by dlouzan** [comment]
@nejch As discussed offline, in my last try locally I downloaded all three upstream `requirements-base.txt`, `requirements-dev.txt` and `requirements-getsentry.txt` and `pip install`-ed all three locally. That at least made dependencies look good locally, but then mypy/pydantic gets in the way.

**Comment by nejch** [comment]
@dlouzan I went with `dev-frozen` so this might be better, but let's see.

File: tests/test_provider.py
Comment (nejch) [inline tests/test_provider.py:1]: We also need to bring this more up to date with https://github.com/getsentry/sentry/blob/master/tests/sentry/auth/providers/google/test_provider.py and then we should be good to go.

File: oidc/provider.py [hunk 1]
```diff
@@ -68,26 +68,20 @@ def get_configure_view(self):
 
     def get_auth_pipeline(self):
         return [
-            OIDCLogin(self.get_client_id(), domains=self.domains),
+            OIDCLogin(domains=self.domains, client_id=self.get_client_id()),
```
Comment (nejch) [inline oidc/provider.py:71]: The changes here are mostly to get a smaller diff against the latest version upstream.

File: pyproject.toml [hunk 2]
```diff
@@ -1,8 +1,12 @@
 [tool.poetry]
 name = "sentry-auth-oidc"
-version = "7.0.0"
+version = "8.0.0"
 description = "OpenID Connect authentication provider for Sentry"
-authors = ["Max Wittig <max.wittig@siemens.com>"]
+authors = [
+    "Max Wittig <max.wittig@siemens.com>",
+    "Diego Louzán <diego.louzan.ext@siemens.com>",
+    "Nejc Habjan <nejc.habjan@siemens.com>",
```
Comment (nejch) [inline pyproject.toml:8]: Diego told me to add this, the takeover is complete now :sweat_smile:

File: tests/test_provider.py
Comment (nejch) [reply tests/test_provider.py:1]: Done, as well as the corresponding conftest to avoid transaction errors.

**Review by max-wittig — APPROVED** [review]

**Comment by dlouzan** [comment]
@nejch Great one! Did we mark it anywhere as breaking?

**Pull request merged**